ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -330,8 +331,13 @@ public void configureBaseApplication(
     config.property(ServerProperties.OUTBOUND_CONTENT_LENGTH_BUFFER, 0);
     config.property(ServerProperties.WADL_FEATURE_DISABLE, true);
 
-    // Registers the REST security extensions
-    securityExtension.register(config, ksqlConfig);
+    // Controls the access to all REST endpoints
+    securityExtension.getAuthorizationProvider().ifPresent(
+        ac -> config.register(new KsqlAuthorizationFilter(ac))
+    );
+
+    // Registers any other security filters (i.e. user context impersonation)
```

Comment:
nit: this comment isn't really helpful, and is a bit confusing (since the impersonation stuff is really a detail of the plugin)

Reply:
Right. I was actually going to remove this call in another PR. My plan is to wrap the impersonation implementation into a call like `extension.getUserContext(principal)`, and make that call from the REST and WS endpoints to get the Kafka and SR clients for that user.

This PR is only authorization so I didn't want to add more functionality.